src/hvps/devices/caen/module.py

```diff
@@ -44,8 +49,12 @@ def number_of_channels(self) -> int:
             int: The number of channels.
         """
         try:
-            response = self._write_command_read_response_module_mon(command="BDNCH")
-            return int(response)
+            method_name = inspect.currentframe().f_code.co_name
```

Comment:
I like the general idea but I think this is very verbose and there is a lot of code duplication, perhaps it could be packed into an auxilary method?